src/Provider/TaxonUrlProvider.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Menu plugin for Sylius.
+ *
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusMenuPlugin\Provider;
+
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class TaxonUrlProvider extends AbstractUrlProvider
+{
+    public const PROVIDER_CODE = 'taxon';
+
+    protected string $code = self::PROVIDER_CODE;
+
+    protected string $icon = 'folder';
+
+    protected int $priority = 100;
+
+    public function __construct(
+        RouterInterface $router,
+        private TaxonRepositoryInterface $taxonRepository,
+    ) {
+        parent::__construct($router);
+    }
+
+    public function getItems(string $locale): array
+    {
+        $taxons = $this->taxonRepository->findBy(['enabled' => true]);
+
+        $items = [];
+        /** @var TaxonInterface $taxon */
+        foreach ($taxons as $taxon) {
+            if ($taxon->isRoot()) {
+                continue;
+            }
+            $taxon->setCurrentLocale($locale); // To have translated name and slug with given locale
+            $items[] = [
+                'name' => $this->getTaxonFullName($taxon),
+                'value' => $this->router->generate('sylius_shop_product_index', ['slug' => $taxon->getSlug(), '_locale' => $locale]),
+            ];
+        }
+
+        usort($items, fn ($itemA, $itemB) => $itemA['name'] <=> $itemB['name']);
+
+        return $items;
+    }
+
+    private function getTaxonFullName(TaxonInterface $taxon): string
+    {
+        $fullName = (string) $taxon->getName();
+        $parent = $taxon->getParent();
+        while (null !== $parent) {
+            /** @var TaxonInterface $parent */
+            if ($parent->isRoot()) {
+                break;
+            }
+            $fullName = $parent->getName() . ' > ' . $fullName;
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** ‏ Maybe add a sprintf here

Reply:
It's outdate now `$taxon->getFullname(' > ')`